cmd/syncthing/summaryservice.go

```diff
@@ -105,14 +105,14 @@ func (c *folderSummaryService) listenForUpdates() {
 					// c.immediate must be nonblocking so that we can continue
 					// handling events.
 
+					c.foldersMut.Lock()
 					select {
 					case c.immediate <- folder:
-						c.foldersMut.Lock()
 						delete(c.folders, folder)
-						c.foldersMut.Unlock()
-
 					default:
+						c.folders[folder] = struct{}{}
```

Comment:
Why this change?

Reply:
Because I had to change a few lines up and this caught my eye as quite wrong: If you want to send a folder summary immediately, but can't, it doesn't make sense to not send one at all.